src/frontend/services/videoService.js

```diff
@@ -13,5 +13,44 @@ const getAllVideos = async () => {
     return { actionResponse , actionSuccess };
 }
 
+const videoInArrayChecker = ( userObjectArray , videosArray , objectArrayName ) => {
+    let activationFlag = false;
+    let customArray = [...videosArray];
+    if( userObjectArray !== [] ) {
+        for( let i = 0; i < userObjectArray.length; i++ ) {
+            for( let j = 0 ; j < customArray.length; j++ ) {
+                if ( userObjectArray[i]._id === customArray[j]._id ) { 
+                    if( objectArrayName === 'like') {
+                        customArray[j].inLikes = true;
+                    } 
+                    if( objectArrayName === 'watchlater' ) {
+                        customArray[j].inWatchlater = true;
+                    }
```

Comment:
Can the find method be used instead of running a logic and setting an attribute to true?

Reply:
hi tejas , I'm still working out a simpler logic for this , 
in this : if an object exists in user's likes/watch later then before videos are displayed make them indicate that the video exits in the likes/watch later (hence the comparison that finds that object exists in both arrays - likes/watchlater & videos)